### PR TITLE
doc: better descriptions for 'create-for-rbac --skip-assignment'

### LINKED ISSUE #8264: More details on --skip-assignment
More details on what --skip-assignment means would be helpful. What is a "default assignment" and why would one want to skip it?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e99d25ff-03e2-e015-37f4-a29b59ec50d4
* Version Independent ID: 1a8e88ed-72c7-fbe7-b0ad-68f23b3946cd
* Content: [az ad sp](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-create-for-rbac)
* Content Source: [src/command_modules/azure-cli-role/azure/cli/command_modules/role/_help.py](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-role/azure/cli/command_modules/role/_help.py)
* Service: **active-directory**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #8264 

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py [hunk 1]
```diff
@@ -68,7 +68,8 @@ def load_arguments(self, _):
     with self.argument_context('ad sp create-for-rbac') as c:
         c.argument('scopes', nargs='+')
         c.argument('role', completer=get_role_definition_name_completion_list)
-        c.argument('skip_assignment', arg_type=get_three_state_flag(), help='do not create default assignment')
+        c.argument('skip_assignment', arg_type=get_three_state_flag(),
+                   help='do not create the default assignment which lets the service principal access resources under current subscription')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py]: The wording here is a little confusing to me. I would suggest:
`Skip creating the default assignment. This allows the service principal to access resources under the current subscription.`  

Also, what if they user supplied `--subscription`?

**Comment by tjprescott** [comment]
@yugangw-msft ping

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py (hunk 1 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py]: The `--subscription` flag has not been wired up yet, which is being tracked by #7901, but the expected behavior is the service principal would be configured to access that subscription.

Regarding suggested text change, to get us on the same page, by default the command configures the access to all resources under a subscription unless you use `--skip-assignment`. So would the following text be better?
   `Skip creating the default assignment, which allows the service principal to access resources under the current subscription`

Comment (tjprescott) [reply src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py]: Got it. This text is clearer.

**Comment by yugangw-msft** [comment]
Done and merge.

**Comment by agianni** [comment]
This is still unclear to me.
The current doc says
 --skip-assignment No-op.
accepted values: false, true
I see several examples that have this parameter but it seems like it may not serve a purpose anymore?
Even in the examples,  I never see it paired with true or false syntax so this is just plain confusing.

Even in the context of this thread the phrase
--skip-assignment
Skip creating the default assignment, which allows the service principal to access resources under the current subscription

It's not clear what the word "which" is referring to.  
What "allows the service principal to access resources under the current subscription"?

Adding --skip-assignment
Omitting --skip-assignment

I am thinking the "which statement" is describing the definition of "default assignment" and not the parameter.
In that case by including -skip-assignment the SP would NOT be granted access to the current subscription until a role and/or scope is assigned.  Please help me understand.

**Pull request merged**